From 587cf711fe3655ee6dce169e05ff0306b15ecd2f Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Thu, 6 Sep 2018 14:04:20 +0200
Subject: [PATCH] Merge pull request #150 in SW_UGW/linux from
 bugfix/DRVLIB_SW-873-null-pointer-access-in-dma_cid_pid_nid_get_chan to
 master

* commit 'faac8694f7ab4a8a6842c6ccc7fe9c27c0a8b539':
  DRVLIB_SW-873: Null pointer access in dma_cid_pid_nid_get_chan
---
 drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c b/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
index 336d32b73985..1380f1ea9ed1 100644
--- a/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/falconmx/cqm.c
@@ -2063,7 +2063,11 @@ static s32 handle_dma_chnl_init(int port, u32 flags)
 	struct cqm_dqm_port_info *p_info;
 	int chan;
 
-	p_info = &dqm_port_info[port];
+	if (find_dqm_port_type(port) == DQM_PON_TYPE)
+		p_info = &dqm_port_info[DQM_PON_START_ID];
+	else
+		p_info = &dqm_port_info[port];
+
 	chan = p_info->dma_ch;
 	if (flags & CBM_PORT_F_DISABLE) {
 		if (chan)
